File: LifeOS/Sources/Core/Models/TimeBlock.swift
import Foundation
import SwiftData

@Model
final class TimeBlock {
    var title: String
    var startTime: Date
    var endTime: Date
    var colorHex: String
    var isCompleted: Bool
    var blockType: String // deepWork, meeting, personal, routine
    
    // Optional link to a task
    var linkedTaskId: String?
    
    init(
        title: String,
        startTime: Date,
        endTime: Date,
        linkedTaskId: String? = nil,
        colorHex: String = "007AFF",
        isCompleted: Bool = false,
        blockType: String = "deepWork"
    ) {
        self.title = title
        self.startTime = startTime
        self.endTime = endTime
        self.linkedTaskId = linkedTaskId
        self.colorHex = colorHex
        self.isCompleted = isCompleted
        self.blockType = blockType
    }
    
    var durationMinutes: Int {
        Int(endTime.timeIntervalSince(startTime) / 60)
    }
    
    var formattedDuration: String {
        let mins = durationMinutes
        if mins < 60 {
            return "\(mins)m"
        } else {
            let hours = mins / 60
            let remaining = mins % 60
            return remaining > 0 ? "\(hours)h \(remaining)m" : "\(hours)h"
        }
    }
}

File: android/app/src/main/java/com/yashlunawat/lifeos/data/models/TimeBlock.kt
package com.yashlunawat.lifeos.data.models

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "time_blocks")
data class TimeBlock(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val title: String,
    val startTime: Long,
    val endTime: Long,
    val linkedTaskId: Long? = null,
    val colorHex: String = "007AFF",
    val isCompleted: Boolean = false,
    val blockType: String = "deepWork" // deepWork, meeting, personal, routine
) {
    val durationMinutes: Int
        get() = ((endTime - startTime) / (1000 * 60)).toInt()

    val formattedDuration: String
        get() {
            val mins = durationMinutes
            return if (mins < 60) {
                "${mins}m"
            } else {
                val hours = mins / 60
                val remaining = mins % 60
                if (remaining > 0) "${hours}h ${remaining}m" else "${hours}h"
            }
        }
}
